components/fxa-client/android/src/main/java/org/mozilla/fxaclient/internal/FirefoxAccount.kt

```diff
@@ -160,6 +160,122 @@ class FirefoxAccount : RustObject {
+     * for the current device.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param displayName The current device display name
+     */
+    fun setDeviceDisplayName(displayName: String) {
+        rustCall { e ->
+            FxaClient.INSTANCE.fxa_set_display_name(validHandle(), displayName, e)
+        }
+    }
+
+    /**
+     * Retrieves the list of the connected devices in the current account, including the current one.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun getDevices(): Array<Device> {
+        val devicesBuffer = rustCall { e ->
+            FxaClient.INSTANCE.fxa_get_devices(validHandle(), e)
+        }
+        try {
+            val devices = MsgTypes.Devices.parseFrom(devicesBuffer.asCodedInputStream()!!)
+            return Device.fromCollectionMessage(devices)
+        } finally {
+            FxaClient.INSTANCE.fxa_bytebuffer_free(devicesBuffer)
+        }
+    }
+
+    /**
+     * Retrieves any pending commands for the current device.
+     * This should be called semi-regularly as the main method of commands delivery (push)
+     * can sometimes be unreliable on mobile devices.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun pollRemoteCommands(): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            FxaClient.INSTANCE.fxa_poll_remote_commands(validHandle(), e)
+        }
+        try {
+            val events = MsgTypes.AccountEvents.parseFrom(eventsBuffer.asCodedInputStream()!!)
+            return AccountEvent.fromCollectionMessage(events)
+        } finally {
+            FxaClient.INSTANCE.fxa_bytebuffer_free(eventsBuffer)
+        }
+    }
+
+    /**
+     * Handle any incoming push message payload coming from the Firefox Accounts
+     * servers.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun handlePushMessage(payload: String): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            FxaClient.INSTANCE.fxa_handle_push_message(validHandle(), payload, e)
+        }
+        try {
+            val events = MsgTypes.AccountEvents.parseFrom(eventsBuffer.asCodedInputStream()!!)
+            return AccountEvent.fromCollectionMessage(events)
+        } finally {
+            FxaClient.INSTANCE.fxa_bytebuffer_free(eventsBuffer)
+        }
+    }
+
+    /**
+     * Ensure the current device "Send Tab" commands has been registered with the server.
+     * This method should be called once per "device lifetime" after the Sync Keys have been
+     * obtained and called again if they change.
```

Comment:
It feels like we should be able to detect internally if the sync keys change (e.g. because our refreshToken gets revoked) so I wonder if there's anything we can do to make this easier on the calling code.

Reply:
I guess we could attempt a registration every time we see a Sync Key pass in front of us since there is only one way to get them, however we'd need a way for apps to opt-out of Send Tab and register the display-uri command (maybe with `Config`?).